### PR TITLE
feat: Eco-Score improvements and change of scale: A+ to F

### PR DESCRIPTION
**Pull request by stephanegigandet**
The Eco-Score has an improved formula:
https://docs.score-environnemental.com/more/changelog#id-19-septembre-2024

There are 2 small changes:
- Previously products that had a non recycled and non non recyclable packaging were capped to Eco-Score B at best. This cap has been removed.
- The country enviromental product index bonus/malus is now only applied if there isn't already a bonus for the production system.

And there is a bigger more visible change:
- the scale is now from A+ to F, instead of A to E, in order to better differentiate products (to avoid having most of the products of the same category have the same Eco-Score).

Note: this is a draft. In particular, the images for the new A+ and F grades are for testing purposes only.

To ease testing of mobile apps etc. this PR has been deployed to the world.openfoodfacts.dev server (login and password: off)

### CODE REVIEW

**Comment by monsieurtanuki** [comment]
> I have strange behaviors in the application, it must be said that we now manage the multi-project. We make a request off then obf, opf and opff, and we stop when the product is found. is that right @monsieurtanuki ?

@teolemon Indeed this is the correct order: food, beauty, pet food, product.

I've just managed to reproduce your screenshots:
* I was happy working on prod with food
* then I switched to `dev`
* then I refreshed a (food/prod) product

The problem seems to be that the product was not found on food/dev. Then of course we're looking for it on beauty/dev, which can be problematic if the beauty/dev server is down or doesn't even exist.

**Comment by monsieurtanuki** [comment]
> The problem seems to be that the product was not found on food/dev. Then of course we're looking for it on beauty/dev, which can be problematic if the beauty/dev server is down or doesn't even exist.

@teolemon That said, a behavior may have to be fixed: if we know for a fact that a product has a given product type (e.g. food), we should check only for this product type and ignore the other ones. Do you want me to implement that behavior?
That said, it would remotely solve your issue, as the product is still not there in food/dev, not very convenient to check ecoscores.

**Comment by stephanegigandet** [comment]
@monsieurtanuki @g123k  Search queries also return an error message on the mobile app when using .dev, even though they seem to be successfull on the server:

[27/Sep/2024:09:52:17 +0000] "POST /cgi/search.pl HTTP/1.0" 200 1431620 "-" "- Smoothie - OpenFoodFacts - 4.15.0+1345 - android+UP1A.231005.007.S911BXXS6CXGA - https://world.openfoodfacts.org/ -  - Smoothie - OpenFoodFacts - 4.15.0+1345 - android+UP1A.231005.007.S911BXXS6CXGA - 6da31523-985c-4bf3-9d23-a58a9d014bf8"

**Comment by monsieurtanuki** [comment]
@stephanegigandet Sure of that? I've just tested on dev for OxF, with my local version and food@dev is OK.
Can you be more specific about the dev query that doesn't work?

| food | beauty |
| -- | -- |
| Screenshot_1727431509 | Screenshot_1727431569 |

| food | beauty |
| -- | -- |
| Screenshot_1727431619 | Screenshot_1727431595 |

**Comment by monsieurtanuki** [comment]
@stephanegigandet I've also tried with real food on food@dev: couldn't find nutella in France, but in the world:
Screenshot_1727431951

**Comment by teolemon** [comment]
po/tags/tags.pot:669: duplicate message definition...
po/tags/tags.pot:660: ...this is the location of the first definition
po/tags/tags.pot:674: duplicate message definition...
po/tags/tags.pot:665: ...this is the location of the first definition
msgfmt: found 2 fatal errors
→ po/tags/tags.pot...contains errors!
po/tags/tags.pot:669: duplicate message definition...
po/tags/tags.pot:660: ...this is the location of the first definition
po/tags/tags.pot:674: duplicate message definition...
po/tags/tags.pot:665: ...this is the location of the first definition
msggrep: found 2 fatal errors

**Comment by teolemon** [comment]
po/common/common.pot:7017: duplicate message definition...
po/common/common.pot:68: ...this is the location of the first definition
po/common/common.pot:7021: duplicate message definition...
po/common/common.pot:72: ...this is the location of the first definition
po/common/common.pot:7025: duplicate message definition...
po/common/common.pot:76: ...this is the location of the first definition
po/common/common.pot:7029: duplicate message definition...
po/common/common.pot:80: ...this is the location of the first definition
po/common/common.pot:7033: duplicate message definition...
po/common/common.pot:84: ...this is the location of the first definition
po/common/common.pot:7037: duplicate message definition...
po/common/common.pot:88: ...this is the location of the first definition
po/common/common.pot:7041: duplicate message definition...
po/common/common.pot:6926: ...this is the location of the first definition
po/common/common.pot:7065: duplicate message definition...
po/common/common.pot:6954: ...this is the location of the first definition
msgfmt: found 8 fatal errors
→ po/common/common.pot...contains errors!
po/common/common.pot:7017: duplicate message definition...
po/common/common.pot:68: ...this is the location of the first definition
po/common/common.pot:7021: duplicate message definition...
po/common/common.pot:72: ...this is the location of the first definition
po/common/common.pot:7025: duplicate message definition...
po/common/common.pot:76: ...this is the location of the first definition
po/common/common.pot:7029: duplicate message definition...
po/common/common.pot:80: ...this is the location of the first definition
po/common/common.pot:7033: duplicate message definition...
po/common/common.pot:84: ...this is the location of the first definition
po/common/common.pot:7037: duplicate message definition...
po/common/common.pot:88: ...this is the location of the first definition
po/common/common.pot:7041: duplicate message definition...
po/common/common.pot:[692](https://github.com/openfoodfacts/openfoodfacts-server/actions/runs/11237806315/job/31241065770?pr=10829#step:4:693)6: ...this is the location of the first definition
po/common/common.pot:7065: duplicate message definition...
po/common/common.pot:6954: ...this is the location of the first definition
msggrep: found 8 fatal errors

File: po/common/common.pot [hunk 1]
```diff
@@ -4811,11 +4811,15 @@ msgctxt "attribute_ecoscore_setting_note"
...
 msgstr ""
 
-# keep %s, it will be replaced by the letter A, B, C, D or E
+# keep %s, it will be replaced by the letter A+, A, B, C, D, E or F
 msgctxt "attribute_ecoscore_grade_title"
 msgid "Eco-Score %s"
 msgstr "Eco-Score %s"
 
+msgctxt "attribute_ecoscore_a_plus_description_short"
+msgid "Very low environmental impact"
```
Comment (teolemon) [inline po/common/common.pot:4820]: GetText will complain about duplicate ids like, and Crowdin will get confused and cross contaminate between the 2 strings

**Review by alexgarel — APPROVED** [review]
LGTM

**Pull request merged**